help/develop/admin/deploy/deploy-docker-swarm.md

```diff
@@ -0,0 +1,119 @@
+---
+title: "Deployment on Docker Swarm"
+---
+
+Datagrok is based on Docker containers, [database](../infrastructure.md#database)
+and [persistent file storage](../infrastructure.md#storage).
+
+As [database](../infrastructure.md#database) Datagrok supports any PostgreSQL database out-of-the-box, including cloud
+solutions for PostgreSQL database, for example [AWS RDS](https://aws.amazon.com/rds/).
+
+For [persistent file storage](../infrastructure.md#storage), Datagrok supports a lot of options, including cloud solutions,
+for example [AWS S3](https://aws.amazon.com/s3/) and Local File System storage.
+
+This document contains instructions to deploy Datagrok using [Docker Swarm](https://docs.docker.com/engine/swarm/)
+on virtual machines or dedicated hosts with Local File System for persistent storage.
+
+More information about Datagrok design and components:
+
+* [Architecture](../architecture.md)
+* [Infrastructure](../infrastructure.md)
+
+## Prerequisites
+
+1. We use native [Docker compose](https://docs.docker.com/compose/) commands to run applications on machines. It
+   simplifies multi-container application development and deployment.
+    1. Download and install the latest version of [Docker Compose](https://docs.docker.com/compose/install/) to your
+       environment (virtual machines, dedicated hosts, etc.)
+
+## Preparations
+
+1. Activate docker swarm mode on your manager node (it should be a datagrok host):
+
+   ```shell
+   docker swarm init
+   ```
+
+   The shell will print a token for adding worker nodes something like that:
+
+   ```shell
+   Swarm initialized: current node (bvz81updecsj6wjz393c09vti) is now a manager.
+
+   To add a worker to this swarm, run the following command:
+
+    docker swarm join \
+    ---token SWMTKN-1-testetsttestetst-qwewsdwqdwdqwddwqdwq XX.XX.XX.XX:2377
+
+   To add a manager to this swarm, run 'docker swarm join-token manager' and follow the instructions.
+   ```
+
+2. Use the received token on worker nodes to join the swarm as described below. You can get
+   token any time by typing in the manager node shell for workers:
+
+   ```shell
+   docker swarm join-token worker
+
+   To add a worker to this swarm, run the following command:
+
+    docker swarm join --token SWMTKN-1-klkjojoijoidowqijd-jdiqjdowqdojwqduudu XX.XX.XX.XX:2377
+   ```
```

Comment:
Remove the part with "You can get token any time by typing in the manager node shell for workers:" it is a piece of extra information here. Lets keep it short and simple